marconi-cardano-indexers/test-lib/Test/Marconi/Cardano/Snapshot.hs

```diff
@@ -29,20 +29,95 @@ import Marconi.Cardano.Indexers.SnapshotBlockEvent (
 import Marconi.Cardano.Indexers.SnapshotBlockEvent qualified as BlockEvent
 import Streaming (Of, Stream)
 import Streaming.Prelude qualified as Stream
-import System.Directory (listDirectory)
+import System.Directory (copyFile, createDirectoryIfMissing, listDirectory)
 import System.FilePath (takeBaseName, (</>))
 
-{- | A 'Snapshot' is an internal representation for a sub-chain which has been
-serialised to disk.
-It stores a ledger state and a stream of blocks. The ledger state is the state
-before the eventual application of the range of blocks in the stream. This
-allows one to simulate the behavior of the blockchain for a specific range of
-blocks which was serialized to disk using the marconi-chain-snapshot executable.
+{- | A snapshot of a sub-chain is a directory containing a series of serialised
+block events and a ledger state of the chain immediately preceding the series of blocks.
+
+'setupSnapshot' is the function which prepares the above data for indexing.
+It creates and returns a stream of block events to be deserialised. The file containing
+the ledger state is copied into Marconi's database.
+
+Ultimately, it provides the necessary data to be able to simulate a partial
+running node.
 -}
-data Snapshot = Snapshot
-  { snapshotPreviousLedgerState :: ExtLedgerStateEvent
-  , snapshotBlockStream :: Stream (Of BlockEvent) IO ()
-  }
+setupSnapshot
+  :: FilePath
+  -- ^ path to the node config file
+  -> FilePath
+  -- ^ directory which contains the serialised events
+  -> FilePath
+  -- ^ directory to be used as the indexer's DB
+  -> IO (Stream (Of BlockEvent) IO ())
+setupSnapshot nodeConfigPath inputDir dbDir = do
+  (ledgerFile, blockFiles) <- findSnapshotFiles inputDir
+  setupLedgerState ledgerFile dbDir
+  getBlockEvents nodeConfigPath blockFiles
+
+-- | Deserialise the files and return the Haskell data for testing.
+deserialiseSnapshot
+  :: FilePath
+  -- ^ path to the node config file
+  -> FilePath
+  -- ^ directory which contains the serialised events
+  -> IO (ExtLedgerStateEvent, Stream (Of BlockEvent) IO ())
+deserialiseSnapshot nodeConfigPath inputDir = do
+  (ledgerFile, blockFiles) <- findSnapshotFiles inputDir
+  blockEvents <- getBlockEvents nodeConfigPath blockFiles
+  ledgerState <- deserialiseLedgerState nodeConfigPath ledgerFile
+  return (ledgerState, blockEvents)
+
+{- | Utility function which searches for the serialised ledger state
+and the serialised block events. The files are expected to be named
+with a specific prefix.
+-}
+findSnapshotFiles
+  :: FilePath
+  -- ^ directory which contains the serialized events
+  -> IO (FilePath, [FilePath])
+findSnapshotFiles inputDir = do
+  files <- fmap (inputDir </>) <$> listDirectory inputDir
+  let blockFiles = filter isBlockEventFile files
+      ledgerStateFile = findLedgerState files
+  when
+    (null blockFiles)
+    (error "Could not find any files containing serialized block events.")
+  return (ledgerStateFile, blockFiles)
```

Comment:
I think we consistenly use `pure` across the codebase, can you change all the `return`?